test/parallel/test-rmdirSync-busy-loop-windows.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+require('../common');
+
+// This test ensures that rmdirSync throws EACCES instead of
+// going into a busy-loop for this scenario:
+// https://github.com/nodejs/node/issues/34580
+
+const assert = require('assert');
+const fs = require('fs');
+const tmpdir = require('../common/tmpdir');
+const { join } = require('path');
+
+function rmdirRecursiveSync() {
+  tmpdir.refresh();
+  const root = fs.mkdtempSync(tmpdir.path);
+
+  const middle = join(root, 'middle');
+  fs.mkdirSync(middle);
+  fs.mkdirSync(join(middle, 'leaf')); // Make `middle` non-empty
+  fs.chmodSync(middle, 0);
```

Comment:
Stripping access permissions appears to have broken our Windows CI on the host the test ran on as subsequent jobs are unable to clean the workspace (as the user doesn't have permissions to remove this directory): https://github.com/nodejs/build/issues/2526

Reply:
Ah yes, that makes sense. `tmpdir.refresh()` uses the rimraf code that's fixed here, and there's not usually a reason for tests to clean up their temp directories. I guess the problem could be solved by having this test do a `tmpdir.refresh()` before exiting. And if we forget to ever remove that workaround, it's not a big deal.

Reply:
Is `tmpdir.refresh()` able to remove a directory where write permissions have been removed?

Reply:
I tried it locally and it does not seem to remove the directory. I think I'll have to change the permission before exiting.

Reply:
I have updated the code to change the permission for the directory to `0o777` before removing it and I don't see any temporary directories after this is run locally. PTAL.

Reply:
If the earlier assertion fails (e.g. as in https://github.com/nodejs/node/pull/36815#issuecomment-763965631) then the reset of the permissions won't occur?

Reply:
No, it does not reset the permissions.